packages/react-log-viewer/src/LogViewer/LogViewer.tsx

```diff
@@ -99,36 +91,37 @@ const LogViewerBase: React.FunctionComponent<LogViewerProps> = memo(
     const [highlightedRowIndexes, setHighlightedRowIndexes] = useState<number[] | null>([]);
     const [currentSearchedItemCount, setCurrentSearchedItemCount] = useState<number>(0);
     const [parsedData, setParsedData] = useState<string[] | null>([]);
-    const [font, setFont] = useState<FontObject>(null);
-
-    const [currentWidth, setCurrentWidth] = useState<number | null>(width || 600);
+    const [lineHeight, setLineHeight] = useState<number>(0);
+    const [charNumsPerLine, setCharNumsPerLine] = useState<number>(0);
     const [resizing, setResizing] = useState(false);
     const [loading, setLoading] = useState(true);
     const [firstMount, setFirstMount] = useState(true);
 
-    const logViewerRef = innerRef || React.useRef<any>();
-    const containerRef = React.useRef<any>();
+    const logViewerRef = innerRef || React.createRef<any>();
+    const containerRef = React.createRef<any>();
     let resizeTimer = null as any;
 
-    React.useEffect(() => {
+    useEffect(() => {
       if (containerRef && containerRef.current && loading) {
         window.addEventListener('resize', callbackResize);
         if (firstMount) {
-          setCurrentWidth((containerRef.current as HTMLDivElement).clientWidth);
           setLoading(false);
-          const dummyText = document.createElement('span');
-          dummyText.className = css(styles.logViewerText);
-          containerRef.current.appendChild(dummyText);
-          const dummyTextStyles = getComputedStyle(dummyText);
-          setFont({
-            fontFamily: dummyTextStyles.fontFamily,
-            fontWeight: dummyTextStyles.fontWeight,
-            fontSize: dummyTextStyles.fontSize,
-            lineHeight: dummyTextStyles.lineHeight
-          });
-          containerRef.current.removeChild(dummyText);
           setFirstMount(false);
         }
+        const dummyText = document.createElement('span');
+        dummyText.className = css(styles.logViewerText);
+        containerRef.current.appendChild(dummyText);
+        const dummyTextStyles = getComputedStyle(dummyText);
+        setLineHeight(parseFloat(dummyTextStyles.lineHeight));
+        const lineWidth = hasLineNumbers
+          ? (containerRef.current as HTMLDivElement).clientWidth - 97
```

Comment:
is 97 the width of the line numbers column? Does it change depending on zoom level? Can you save it to a variable?

Reply:
@jschuler Nice catch! It changes with the base font. I will update this, thanks!